docs/non_native_field_arithmetic.md

```diff
@@ -0,0 +1,165 @@
+# Non-Native Field Arithmetic
+
+**References:**
+
+- [Aztec Emulated Field and Group Operations for Efficient Large Number Arithmetic](https://hackmd.io/@kullervo/cryptographyExplorations/%2F%40kullervo%2FbignumberCircuits)
+- [Non-Native Field Arithmetic](https://hackmd.io/@JkY-zACaSqerTtn_UwFjKg/SJZw6x75o)
+- [Field selection for recursive SNARKs](https://medium.com/delendum/field-selection-for-recursive-snarks-726ad56c3a3c)
+- [Foreign field arithmetic](https://www.youtube.com/watch?v=IBeo5epQoYo)
+
+## Motivation
+
+Sometimes, we want to represent modulo arithmetic with prime modulus $p$ over our prime finite field $\mathbb{F}_n$ . It's called
+non-native field arithmetic (or some other names like *foreign field, mismatched field, non-aligned field, wrong field*). Below are a
+few use cases:
+
+- **For ZK-Rollup after EIP-4844:** After [EIP-4844](eip_4844.md), this operation is required for proof systems that do not
+  use [BLS12-381](../terms/bls12-381.md) modulus to deal with blob transaction.
```

Comment:
```suggestion
  use [BLS12-381](../terms/bls12-381.md) modulus to deal with blob transactions.
```

Reply:
"blob transaction" is a transaction type, not a count, so it has no "s".